src/utils.jl

```diff
@@ -97,3 +113,88 @@ function shuffle_split_stratified(
     end
     return inds1, inds2
 end
+
+_sample_dims(data::AbstractVector) = Colon()
+_sample_dims(data::AbstractArray{<:Any,3}) = (1, 2)
+
+"""
+    fold([f,] x::AbstractVector)
+    fold([f,] x::AbstractArray{<:Any,3}; dims=(1, 2))
+
+Compute the absolute deviation of `x` from `f(x)`, where `f` defaults to `Statistics.median`.
+
+`f` is generally a measure of central tendency. `dims` are the dimensions over which the
+estimator `f` reduces and are passed as kwargs to `f`.
+"""
+fold(f, data; dims=_sample_dims(data)) = abs.(data .- f(data; dims=dims))
```

Comment:
Dispatch on `data` since according to the docstring `dims` is not supported for `data::AbstractVector`?

Moreover, as Julia does not specialize on functions, we want to use a type parameter here, I assume:
```suggestion
function fold(f::F, data::AbstractVector) where {F}
    return abs.(data .- f(data))
end
function fold(f::F data::AbstractArray{<:Any,3}; dims=(1,2)) where {F}
    return abs.(data .- f(data; dims=dims))
end
```